### ISSUE TITLE
LoggingLevel::Off should Paint::disable(); to enable non-colored output on other slog modules

### ISSUE THREAD

**Issue by sebosp** [SergioBenitez/Rocket#1138]
Hi,

Thanks for the awesome project. I am currently trying to get JSON logs out of Rocket to send them later to ES/Athena.

Currently, I could `attach()` a `SyncLogger` to the mounted routes to get JSON messages out of Rocket (using `rocket_slog::SyncLogger` and `slog_json`).
To get the output clean, I used `config.log_level(LoggingLevel::Off)`, but then, when I see the output in the terminal, I see lines like:
```text
{"msg":"response","level":"INFO","route":"\u001b[32mGET\u001b[0m \u001b[34m/version\u001b[0m \u001b[36m(\u001b[0m\u001b[35mversion\u001b[0m\u001b[36m)\u001b[0m"}
```

It seems to me this is due to the early return of this function, `try_init()`. Should this function also include the `Paint::disable();` before returning false?
https://github.com/SergioBenitez/Rocket/blob/da7e022f990e0b8e8201b0a359a43104686ff1a4/core/lib/src/logger.rs#L149

Or could this be done in another way?
Some code context:
```rust
#[macro_use(debug)]
extern crate slog;
extern crate rocket_slog;
extern crate slog_async;
extern crate slog_json;

use rocket_slog::SlogFairing;
use slog::Drain;
use std::env;

#[get("/version")]
fn version(log: SyncLogger) -> &'static str {
    debug!(log, "Some log message");
    env!("CARGO_PKG_VERSION")
}

fn main() {
    let config = Config::build(Environment::Development)
        .log_level(LoggingLevel::Off) // disables logging, but need to call Paint::disable();
        .finalize()
        .unwrap();
    let drain = slog_json::Json::new(std::io::stdout())
        .set_pretty(false)
        .add_default_keys()
        .build()
        .fuse();
    let drain = slog_async::Async::new(drain).build().fuse();
    let fairing: SlogFairing::new(slog::Logger::root(
        drain,
        o!("testkey" => "val"),
    ));
    rocket::custom(config)
        .mount("/", routes![version])
        .attach(fairing)
        .launch();
```

One workaround is to set the `LoggingLevel::Critical`, which shows the messages without the `[b` color escapes, but then the output from Rocket plain text and the JSON messages appear in the same context.

**Comment by jebrosen** [comment]
This is a bit of an odd situation, but I think it makes sense to always disable `Paint` if `LoggingLevel` is `Off`. In that situation, either nothing will ever get logged and disabling `Paint` is harmless or something other than Rocket will log messages and we shouldn't assume color codes are safe.

**Comment by SergioBenitez** [comment]
Let's fix this as part of #21.
